File: benchmark/c/cpw/count_possible_paths_top_left_bottom_right_nxm_matrix_3/count_possible_paths_top_left_bottom_right_nxm_matrix_3.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int count_possible_paths_top_left_bottom_right_nxm_matrix_3 ( int m, int n ) ;
int count_possible_paths_top_left_bottom_right_nxm_matrix_3 ( int m, int n ) {
  int path = 1;
  for ( int i = n;
  i < ( m + n - 1 );
  i ++ ) {
    path *= i;
    path /= ( i - n + 1 );
  }
  return path;
}

File: translation_result/else_bms/fail/c/cpw/count_possible_paths_top_left_bottom_right_nxm_matrix_3/gemini/BaseRepair/count_possible_paths_top_left_bottom_right_nxm_matrix_3.rs

fn count_possible_paths_top_left_bottom_right_nxm_matrix_3(m: i32, n: i32) -> i32 {
    if m < 0 || n < 0 {
        return 0;
    }

    let mut path = 1;
    for i in n..(m + n - 1) {
        if i > i32::MAX / path {
            return 0;
        }
        path *= i;
        path /= i - n + 1;
    }
    path
}
